arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/stream/callbacks.kt

```diff
@@ -121,7 +121,7 @@ fun <A> Stream.Companion.cancellable(@BuilderInference f: suspend EmitterSyntax<
     val cancel = Promise<CancelToken>()
 
     Stream.bracketCase({
-      ForkAndForget { emitterCallback(f, cancel, error, q) }
+      ForkAndForget(EmptyCoroutineContext) { emitterCallback(f, cancel, error, q) }
```

Comment:
What's the reason to fork here if we staying on the same context? 🤔

Reply:
`Fork`'ing on `EmptyCoroutineContext` doesn't actually fork, and the concern is still left to the user like @JorgeCastilloPrz linked in the snippet of the issue? In that case, there is no need to `ForkAndForget` here.

`lifecycle.addObserver` is the same as the use-case of a websocket.

Reply:
The only difference is that Android will crash if it detects you attach the observer in a thread that's not the Main one

Reply:
FYI tests failed after removing `ForkAndForget`, I'm unsure if it's a test problem or that it is really necessary since they passed before.

![image](https://user-images.githubusercontent.com/3294640/96743620-7dcc7080-13c4-11eb-8f55-de7140ab6a14.png)